labs/RivePlayer/src/RivePlayer.cs

```diff
@@ -0,0 +1,255 @@
+///   </rive:RivePlayer>
+///
+/// Rive state machines: https://rive.app/features#state-machine
+/// State machine inputs: https://help.rive.app/editor/state-machine#inputs
+/// </summary>
+public sealed partial class RivePlayer
+{
+    // Continuously invalidates the panel for repaint.
+    private AnimationTimer? _animationTimer;
+
+    public RivePlayer()
+    {
+        this.DefaultStyleKey = typeof(RivePlayer);
+
+        this.StateMachineInputCollection.SetRivePlayer(this);
+        this.Loaded += OnLoaded;
+        this.PointerPressed +=
+            (object s, PointerRoutedEventArgs e) => HandlePointerEvent(_scene.PointerDown, e);
+        this.PointerMoved +=
+            (object s, PointerRoutedEventArgs e) => HandlePointerEvent(_scene.PointerMove, e);
+        this.PointerReleased +=
+            (object s, PointerRoutedEventArgs e) => HandlePointerEvent(_scene.PointerUp, e);
+    }
+
+    private void OnXamlRootChanged(bool isHostVisible)
+    {
+        if (_animationTimer == null)
+        {
+            return;
+        }
+        if (isHostVisible)
+        {
+            _animationTimer.Start();
+        }
+        else
+        {
+            _animationTimer.Stop();
+        }
+    }
+
+    // State machine inputs to set once the current async source load finishes.
+    // The null-ness of this object also tells us whether an async load operation is currently running.
+    private List<Action>? _deferredSMInputsDuringAsyncSourceLoad = null;
+
+    private async void LoadSourceFileDataAsync(string uriString, int sourceToken)
+    {
+        if (Uri.TryCreate(uriString, UriKind.Absolute, out var uri))
+        {
+            Stream? stream = null;
+            if (uri.Scheme == "http" || uri.Scheme == "https")
+            {
+                try
+                {
+                    HttpResponseMessage response = await new HttpClient().GetAsync(uri);
+                    if (response.IsSuccessStatusCode)
+                    {
+                        stream = await response.Content.ReadAsStreamAsync();
+                    }
+                }
+                catch (HttpRequestException e)
+                {
+                    // TODO: Load a 404 file?
+                    Console.WriteLine("Failed to connect to " + uri.ToString());
+                    Console.WriteLine(e.ToString());
+                }
+            }
+            else if (uri.Scheme == "ms-appx")
+            {
+                var file = await StorageFile.GetFileFromApplicationUriAsync(uri);
+                if (file != null && sourceToken == _currentSourceToken)
+                {
+                    IBuffer buffer = await FileIO.ReadBufferAsync(file);
+                    stream = buffer.AsStream();
+                }
+            }
+            if (stream != null && sourceToken == _currentSourceToken)
+            {
+                byte[] data = new byte[stream.Length];
+                stream.Read(data, 0, data.Length);
```

Comment:
This is doing blocking IO on the UI thread, consider using `ReadAsync` passing the token instead.